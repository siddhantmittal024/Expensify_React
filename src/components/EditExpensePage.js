import { Switch, BrowserRouter, Route, Link, NavLink } from "react-router-dom";
import React from "react";

const EditExpensePage = (props) => {
  return (
    <div>This is from my edit expense page of id {props.match.params.id}</div>
  );
};

export default EditExpensePage;
