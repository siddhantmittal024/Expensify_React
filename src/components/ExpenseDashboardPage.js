import { Switch, BrowserRouter, Route, Link, NavLink } from "react-router-dom";
import React from "react";

const ExpenseDashboardPage = () => <div>This is from my DashboardPage.</div>;

export default ExpenseDashboardPage;
