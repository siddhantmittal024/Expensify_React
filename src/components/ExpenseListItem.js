// Export a stateless functional component
// description, amount, createdAt
import React from 'react';
import {Link} from 'react-router-dom';
import momemt from 'moment';
import numeral from 'numeral';

const ExpenseListItem  = ({description,id,amount,createdAt}) => (
    <div>
        <Link to={`/edit/${id}`}>
        <h3>{description}</h3>
        </Link>
        <p>
            {numeral(amount).format('$0,0.00')}
             - 
            {momemt(createdAt).format('MMMM Do, YYYY')}
        </p>
    </div>
);


export default ExpenseListItem; 