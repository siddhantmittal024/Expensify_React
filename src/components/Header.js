import { Switch, BrowserRouter, Route, Link, NavLink } from "react-router-dom";
import React from "react";

const Header = () => (
  <header>
    <h1>Expensify</h1>
    <NavLink to="/" activeClassName="is-active" exact={true}>
      Dashboard
    </NavLink>
    <NavLink to="/create" activeClassName="is-active">
      Create
    </NavLink>
    <NavLink to="/help" activeClassName="is-active">
      ExpenseHelp
    </NavLink>
  </header>
);

export default Header;
