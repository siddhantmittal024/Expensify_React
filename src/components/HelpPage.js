import { Switch, BrowserRouter, Route, Link, NavLink } from "react-router-dom";
import React from "react";

const HelpPage = () => <div>This is from my help page.</div>;

export default HelpPage;
