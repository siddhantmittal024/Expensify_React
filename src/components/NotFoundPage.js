import { Switch, BrowserRouter, Route, Link, NavLink } from "react-router-dom";
import React from "react";

const NotFoundPage = () => (
  <div>
    404! - <Link to="/">Go Home</Link>
  </div>
);

export default NotFoundPage;
