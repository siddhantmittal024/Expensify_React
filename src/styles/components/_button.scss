.button {
  background: $blue;
  color: white;
  border: none;
  padding: $s-size;
  display: inline-block;
  font-weight: 300;
  font-size: 1.8rem;
  text-decoration: none;
  line-height: 1;
  border-radius: 3px;
}

.button--link {
  margin: 2rem;
  background-color: rgba(white, 0.05);
  border-bottom: 2px solid lighten(rgba(#364051, 0.9), 20%);
  margin: 0;
  &:hover {
    background-color: rgba(white, 0.1);
  }
}

.button--secondary {
  background: #888;
}
