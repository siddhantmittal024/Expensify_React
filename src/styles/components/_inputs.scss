.text-input {
  border: 1px solid #cacccd;
  height: 50px;
  padding: $s-size;
  font-size: 1.8rem;
  font-weight: 300;
}

.select {
  @extend .text-input;
}

.textarea {
  @extend .text-input;
  height: 10rem;
}
