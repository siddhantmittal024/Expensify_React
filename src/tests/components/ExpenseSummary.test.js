import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseSummary } from '../../components/ExpenseSummary';

test('should correctly render ExpensesSummary with 1 expense', () => {
  const wrapper = shallow(<ExpenseSummary expenseCount={1} expenseTotal={235} />);
  expect(wrapper).toMatchSnapshot();
});

test('should correctly render ExpensesSummary with multiple expenses', () => {
  const wrapper = shallow(<ExpenseSummary expenseCount={23} expenseTotal={23512340987} />);
  expect(wrapper).toMatchSnapshot();
});
