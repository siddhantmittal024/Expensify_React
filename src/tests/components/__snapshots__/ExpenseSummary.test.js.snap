// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render ExpensesSummary with 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $235.00
  </h1>
</div>
`;

exports[`should correctly render ExpensesSummary with multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    23
     
    expenses
     totalling 
    $23,512,340,987.00
  </h1>
</div>
`;
